src/odemis/util/img.py

```diff
@@ -1201,3 +1201,41 @@ def assembleZCube(images, zlevels):
         ret = DataArray(ret, metadata3d)
 
         return ret
+
+
+def apply_flood_fill(array, start):
+    """
+    Flood fills a 2-Dimensional numpy array with truth values from a given start position with the 4-connected method.
+    The input array contains a shape represented with truth values for the borders of the shape.
+
+    :param array: (ndarray(bool)) binary array of size mxn containing the shape to be filled
+    :param start: (tuple(int, int)) position from which to start flood fill
+    :return: (ndarray(bool)) array of size mxn containing the filled shape, with filled values True
+    """
+    if start >= array.shape:
+        raise ValueError(f"Start position of {start} does not lie within the array.")
+
+    array = array.copy()
+    max_area = array.shape[0] * array.shape[1]
+    queue = [start]
+    overflow_counter = 0
+    while len(queue) > 0:
+        # To prevent the possibility of an infinite loop count the number of iterations. If it is larger than the
+        # total number of values in the array raise an error since this should not be possible.
+        if overflow_counter > max_area:
+            raise ValueError(f"Number of loop iterations is higher than maximum iterations possible ({max_area}) in "
+                             f"array of shape: {array.shape}")
+
+        row, col = queue.pop(0)  # Set the current pixel to the first element of the queue and remove it from the queue
+        if not array[row, col]:
```

Comment:
If I understood correctly, I think the below code can be optimized reference: https://www.baeldung.com/cs/flood-fill-algorithm#2-naive-implementation